baybe/constraints/discrete.py

```diff
@@ -77,6 +127,29 @@ def get_invalid(self, data: pd.DataFrame) -> pd.Index:  # noqa: D102
 
         return data.index[mask_bad]
 
+    def get_polars_expression(self, df: pl.LazyFrame) -> pl.Expr:
+        """Translate the conditions to polars expression for filtering.
+
+        Args:
+            df: A polars lazy frame of all parameters.
+
+        Returns:
+            The Polars Expr object to pass as an argument to filter().
+
+        """
+        op = _threshold_operators[self.condition.operator]
+        # Identify columns present in both self.parameters and df.columns
+        cols = [c for c in self.parameters if c in df.columns]
+
+        # Check if all the columns in self.parameters are present in df.columns
+        if set(cols) == set(self.parameters):
```

Comment:
I removed all these checks because we no longer filter during join (as you described [here](https://github.com/emdgroup/baybe/pull/252#discussion_r1646679946))